test/AutoDiff/control_flow.swift

```diff
@@ -1,79 +1,277 @@
-// RUN: %target-swift-frontend -emit-sil -verify -Xllvm -differentiation-enable-control-flow %s | %FileCheck %s -check-prefix=CHECK-DATA-STRUCTURES
-// RUN: %target-swift-frontend -emit-sil -verify -Xllvm -differentiation-enable-control-flow %s | %FileCheck %s -check-prefix=CHECK-SIL
+// RUN: %target-run-simple-swift-control-flow-differentiation
+// REQUIRES: executable_test
 
-// Test cond. A simple if-diamond.
+import StdlibUnittest
 
-@differentiable
-@_silgen_name("cond")
-func cond(_ x: Float) -> Float {
-  if x > 0 {
+var ControlFlowTests = TestSuite("ControlFlow")
+
+ControlFlowTests.test("Conditionals") {
+  func cond1(_ x: Float) -> Float {
+    if x > 0 {
+      return x * x
+    }
     return x + x
   }
-  return x - x
+  expectEqual(8, gradient(at: 4, in: cond1))
+  expectEqual(2, gradient(at: -10, in: cond1))
+
+  func cond2(_ x: Float) -> Float {
+    let y: Float
+    if x > 0 {
+      y = x * x
+    } else if x == -1337 {
+      y = 0
+    } else {
+      y = x + x
+    }
+    return y
+  }
+  expectEqual(8, gradient(at: 4, in: cond2))
+  expectEqual(2, gradient(at: -10, in: cond2))
+  expectEqual(0, gradient(at: -1337, in: cond2))
+
+  func cond2_var(_ x: Float) -> Float {
+    var y: Float = x
+    if x > 0 {
+      y = y * x
+    } else if x == -1337 {
+      y = x // Dummy assignment; shouldn't affect computation.
+      y = x // Dummy assignment; shouldn't affect computation.
+      y = 0
+    } else {
+      y = x + y
+    }
+    return y
+  }
+  expectEqual(8, gradient(at: 4, in: cond2_var))
+  expectEqual(2, gradient(at: -10, in: cond2_var))
+  expectEqual(0, gradient(at: -1337, in: cond2_var))
+
+  func cond3(_ x: Float, _ y: Float) -> Float {
+    if x > 0 {
+      return x * y
+    }
+    return y - x
+  }
+  expectEqual((5, 4), gradient(at: 4, 5, in: cond3))
+  expectEqual((-1, 1), gradient(at: -3, -2, in: cond3))
+
+  func cond_tuple(_ x: Float) -> Float {
+    let y: (Float, Float) = (x, x)
+    return y.0 + y.1
+  }
+  expectEqual(8, gradient(at: 4, in: cond2))
```

Comment:
Oops, I messed up the function name here, and for `cond_tuple_var` below.
```suggestion
  expectEqual(8, gradient(at: 4, in: cond_tuple))
```
Will fix and retrigger CI.